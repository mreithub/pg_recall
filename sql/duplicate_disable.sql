--
-- Creates a simple table, enables logging but then disables it twice (which should raise an exception)
--

BEGIN;

CREATE TABLE config (
	key VARCHAR(100) PRIMARY KEY,
	value TEXT
);

SELECT recall_enable('config', '2 months');

SELECT recall_disable('config');

-- ok, everything should have worked as planned so far, but this next call should trigger an exception
SELECT recall_disable('config');

ROLLBACK;
