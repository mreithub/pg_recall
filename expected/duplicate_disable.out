--
-- Creates a simple table, enables logging but then disables it twice (which should raise an exception)
--
BEGIN;
CREATE TABLE config (
	key VARCHAR(100) PRIMARY KEY,
	value TEXT
);
SELECT recall.enable('config', '2 months');
 enable 
--------
 
(1 row)

SELECT recall.disable('config');
NOTICE:  view "config_past" does not exist, skipping
CONTEXT:  SQL statement "DROP VIEW IF EXISTS config_past"
PL/pgSQL function recall.disable(regclass) line 22 at EXECUTE
 disable 
---------
 
(1 row)

-- ok, everything should have worked as planned so far, but this next call should trigger an exception
SELECT recall.disable('config');
ERROR:  The table "config" is not managed by pg_recall
ROLLBACK;
